### PR TITLE
Add DINO ImageNet ResNet50 benchmark

### PR DESCRIPTION
**Pull request by guarin**
## Changes

* Add DINO ImageNet ResNet50 benchmark

## Results

```
imagenet
	max val_online_cls_top1: 	0.6590800285339355
	max val_online_cls_top5: 	0.8774999976158142
	knn val_top1: 			0.49946001172065735
	knn val_top5: 			0.7870200276374817
	max linear val_top1: 		0.6817200183868408
	max linear val_top5: 		0.8794599771499634
	max finetune val_top1: 		0.7249799966812134
	max finetune val_top5: 		0.9077799916267395
```

KNN results are lower than in the paper (49% vs. 65.7%) as we use a different number of nearest neighbors (200 vs. 20) and fewer training epochs (100 vs 300). KNN accuracy is slightly better than what we got for SwAV (49.9% vs 49.5%). 

Linear validation results are also lower than in the paper (68.2% vs. 75.3%) as we use fewer pretraining epochs (100 vs. 300) and fewer linear training epochs (90 vs 100). Looking at Table 2 (below) from the paper we should get results that are on par with SwAV:
Screen Shot 2023-06-13 at 15 34 55

In our case, DINO is slightly better than SwAV (68.2% vs 67.2%), see [benchmark results](https://github.com/lightly-ai/lightly#imagenet).

**Note**: Training DINO took significantly more time than SwAV/SimCLR and requires also slightly more memory (had to decrease batch size to 64/device).

Tensorboard logs: https://tensorboard.dev/experiment/DvKHX9sNSWWqDrRksllPLA

### CODE REVIEW

File: benchmarks/imagenet/resnet50/dino.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        global_views = torch.cat(views[:2])
+        local_views = torch.cat(views[2:])
+
+        teacher_features = self.forward(global_views).flatten(start_dim=1)
+        teacher_projections = self.projection_head(teacher_features)
+        student_projections = torch.cat(
+            [self.forward_student(global_views), self.forward_student(local_views)]
+        )
+
+        loss = self.criterion(
```
Comment (philippmwirth) [inline benchmarks/imagenet/resnet50/dino.py:80]: use keywords please

File: benchmarks/imagenet/resnet50/dino.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                    / self.trainer.max_epochs
+                    * 10
+                ),
+                max_epochs=self.trainer.estimated_stepping_batches,
+            ),
+            "interval": "step",
+        }
+        return [optimizer], [scheduler]
+
+    def configure_gradient_clipping(
```
Comment (philippmwirth) [inline benchmarks/imagenet/resnet50/dino.py:150]: where is this coming from? I didn't see anything in the paper

**Review by philippmwirth — APPROVED** [review]

File: benchmarks/imagenet/resnet50/dino.py (hunk 2 above)
Comment (guarin) [reply benchmarks/imagenet/resnet50/dino.py:150]: Default is 3: https://github.com/facebookresearch/dino/blob/7c446df5b9f45747937fb0d72314eb9f7b66930a/main_dino.py#L87
